### PR TITLE
Enforce RuboCop Styling Rules through Travis

### PR DESCRIPTION
**Pull request by oakbani**
**Issue:** Though, [contributing.md](https://github.com/optimizely/ruby-sdk/blob/master/CONTRIBUTING.md) says, to lint code with Rubocop rules and we also have a rubocop.yml file in the repo, the rules are not enforced. 
🖊 Rubocop reported a total of 1136 offences
🖊 Added Rubocop linter command to Travis config file to ensure all the future code is linted before submission to repo. See how Travis reports issues : [here](https://travis-ci.org/msohailhussain/ruby-sdk/builds/287579431)
🖊 Auto and Manually corrected all Rubo Cop reported offences
🖊 Generated a .rubocop_todo.yml file to ignore a small number of offences and ignore non-ruby files
🖊 Rubocop auto added +# frozen_string_literal: true this line to every file. See [here](https://www.lucascaton.com.br/2016/01/19/what-is-frozen_string_literal-in-ruby/), why it is doing so.

### CODE REVIEW

**Comment by msohailhussain** [comment]
owais I checked it, let's discuss tonight.

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+# Offense count: 1
+Lint/HandleExceptions:
+  Exclude:
+    - 'Rakefile'
+
+# Offense count: 2
+Lint/UselessAssignment:
+  Exclude:
+    - 'lib/optimizely/decision_service.rb'
```
Comment (aliabbasrizvi) [inline .rubocop_todo.yml]: Why is it excluded?

File: lib/optimizely.rb [hunk 2]
```diff
@@ -39,7 +39,8 @@ class Project
     attr_reader :event_dispatcher
     attr_reader :logger
 
-    def initialize(datafile, event_dispatcher = nil, logger = nil, error_handler = nil, skip_json_validation = false, user_profile_service = nil)
+    def initialize(datafile, event_dispatcher = nil, logger = nil, error_handler = nil, skip_json_validation = false,
```
Comment (aliabbasrizvi) [inline lib/optimizely.rb]: nit. Looks awkward with just 1 parameter on the next line. Perhaps do an equal split.

File: lib/optimizely/bucketer.rb [hunk 3]
```diff
@@ -1,3 +1,4 @@
+# frozen_string_literal: true
```
Comment (aliabbasrizvi) [inline lib/optimizely/bucketer.rb]: Seems unnecessary.

File: lib/optimizely/condition.rb [hunk 4]
```diff
@@ -1,3 +1,4 @@
+# frozen_string_literal: true
```
Comment (aliabbasrizvi) [inline lib/optimizely/condition.rb]: same as above

File: lib/optimizely/condition.rb [hunk 5]
```diff
@@ -1,3 +1,4 @@
...
 #
 #    Copyright 2016, Optimizely and contributors
```
Comment (aliabbasrizvi) [inline lib/optimizely/condition.rb]: Update to 2016-2017

File: lib/optimizely/condition.rb [hunk 6]
```diff
@@ -27,7 +28,7 @@ class ConditionEvaluator
       ConditionalOperatorTypes::AND,
       ConditionalOperatorTypes::OR,
       ConditionalOperatorTypes::NOT
-    ]
+    ].freeze
```
Comment (aliabbasrizvi) [inline lib/optimizely/condition.rb]: Is this for making this constant?

File: lib/optimizely/bucketer.rb (hunk 3 above)
Comment (aliabbasrizvi) [reply lib/optimizely/bucketer.rb]: Is it to make it Ruby 2 compatible?

File: spec/spec_params.rb [hunk 7]
```diff
@@ -51,15 +52,15 @@ module OptimizelySpec
         'endOfRange' => 5000
       }, {
         'entityId' => '111129',
-        'endOfRange' => 10000
+        'endOfRange' => 10_000
```
Comment (aliabbasrizvi) [inline spec/spec_params.rb]: Why? I read it is to improve readability. I'd prefer keeping it as it was.

File: lib/optimizely/bucketer.rb (hunk 3 above)
Comment (aliabbasrizvi) [reply lib/optimizely/bucketer.rb]: Can you see if this comment can be at the end of license text?

Comment (aliabbasrizvi) [reply lib/optimizely/bucketer.rb]: Looks like it has to be first line: https://stackoverflow.com/questions/37799296/what-does-the-comment-frozen-string-literal-true-do

**Pull request closed without merging**